src/model/openthaigpt_pretraining_model/llama/model.py

```diff
@@ -1,14 +1,215 @@
-class ExampleModel:
-    def __init__(self):
-        pass
-
-    def predict(self, x: int) -> int:
-        return x**3
-
-
-class ExampleModelModified:
-    def __init__(self):
-        pass
-
-    def predict(self, x: int) -> int:
-        return x * x * x
+from typing import Optional
+from dataclasses import dataclass
+import math
+
+import torch
+from torch import nn
+import torch.nn.functional as F  # noqa: N812
+
+from torch.nn.functional import scaled_dot_product_attention as torch_attention
+from xformers.components.attention.core import (
+    scaled_dot_product_attention as xformer_attention,

```

Comment:
Please use this one instead: https://facebookresearch.github.io/xformers/components/ops.html

Reply:
done